e2e/test/iothub/service/ExportDevicesTests.cs

```diff
@@ -170,14 +170,14 @@ public async Task DevicesClient_ExportDevices(StorageAuthenticationType storageA
 
             var sw = Stopwatch.StartNew();
 
-            while (!jobProperties.IsFinished)
+            while (!exportJobProperties.IsFinished)
             {
                 try
                 {
-                    jobProperties = await serviceClient.Devices.ExportAsync(jobProperties).ConfigureAwait(false);
-                    if (!string.IsNullOrWhiteSpace(jobProperties.FailureReason))
+                    exportJobProperties = (ExportJobProperties)await serviceClient.Devices.ExportAsync(exportJobProperties).ConfigureAwait(false);
```

Comment:
We have to cast it?

Reply:
we do have to cast it as we are using some of the class specific properties.